src/ocm/components/fetching/fetchEvents.ts

```diff
@@ -1,15 +1,28 @@
 import { EventListFetchProps } from '../../../common';
 import { handleApiError } from '../../api';
 import { EventsAPI } from '../../services/apis';
+import InfraEnvsService from '../../services/InfraEnvsService';
 
 export const onFetchEvents: EventListFetchProps['onFetchEvents'] = async (
   props,
   onSuccess,
   onError,
 ) => {
   try {
-    const { data } = await EventsAPI.list(props);
-    onSuccess(data);
+    const { data: eventList } = await EventsAPI.list(props);
+
+    if (!props.hostId) {
+      const infraEnvId = await InfraEnvsService.getInfraEnvId(props.clusterId);
+      const { data } = await EventsAPI.list({ infraEnvId: infraEnvId });
+
+      eventList.push(
+        ...data.filter(
+          (event) => !eventList.find((e) => e.eventTime + e.message === event.eventTime + e.message),
+        ),
```

Comment:
IIUC there is a chance we get duplicated events???

Reply:
Between the cluster and infraEnv events, we get more duplicates than unique ones. @jkilzi

Reply:
@osherdp can you please or somebody from the BE take a look at this part?
We are doing this in order to remove duplicated events coming from the cluster and the infra-env...

Reply:
@slavie maybe can take a look
she knows this part well

Reply:
sorry, mentioned the wrong person
@slaviered

Reply:
The filtering should be done on the BE side. Please open a bug over this issue.